pkg/content/extensions/extensions.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extensions
+
+import (
+	"io"
+	"io/fs"
+
+	"github.com/GoogleContainerTools/kpt/pkg/location"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+)
+
+type Content interface {
+	io.Closer
+}
+
+type ChangeCommitter interface {
+	Content
+	CommitChanges() (location.Location, error)
+}
+
+type FileSystemProvider interface {
+	Content
+	ProvideFileSystem() (filesys.FileSystem, string, error)
+}
+
+type FSProvider interface {
+	Content
+	ProvideFS() (fs.FS, error)
+}
+
+type ReaderProvider interface {
+	Content
+	ProvideReader() (kio.Reader, error)
+}
+
+type RealPathProvider interface {
```

Comment:
TBH - this can probably be removed. It was for the `kpt pkg diff` case --- the idea being when you _must_ have a real OS file system path you could use the existing one for content that was already on disk. (like dir and git that is based on exec git clone)

In practice --- the diff command actually must copy from src to staging location.Dir for other reasons (because prepping for diff is destructive). So I'll just remove this extension.